src/app/@ansyn/imagery/model/base-imagery-map.ts

```diff
@@ -23,22 +23,28 @@ export abstract class BaseImageryMap<T = any> {
 	readonly defaultMapSource?: string;
 
 	public positionChanged: EventEmitter<ICaseMapPosition> = new EventEmitter<ICaseMapPosition>();
+	public tilesLoadProgressEventEmitter: EventEmitter<IMapProgress> = new EventEmitter<IMapProgress>();
+	public tilesLoadErrorEventEmitter: EventEmitter<IMapErrorMessage> = new EventEmitter<IMapErrorMessage>();
 	public mapObject: T;
+	public backgroundMapObject: T;
 
 	abstract getCenter(): Observable<Point>;
 
 	abstract setCenter(center: Point, animation: boolean): Observable<boolean>;
 
 	abstract toggleGroup(groupName: string, newState: boolean);
 
-	abstract initMap(element: HTMLElement, shadowElement: HTMLElement, layers?: any, position?: ICaseMapPosition): Observable<boolean>;
+	abstract initMap(element: HTMLElement, shadowNorthElement: HTMLElement, shadowDoubleBufferElement: HTMLElement, layers?: any, position?: ICaseMapPosition): Observable<boolean>;
+
+	initMapSubscriptions(): void {
```

Comment:
add remark that this function is doing Auto-subscribe

Reply:
done